src/core/src/op/non_max_suppression.cpp

```diff
@@ -805,15 +805,12 @@ void op::v5::NonMaxSuppression::validate_and_infer_types() {
     validate();
 
     if (boxes_ps.rank().is_static() && scores_ps.rank().is_static() && get_input_size() > 2) {
-        const auto num_boxes_boxes = boxes_ps[1];
-        if (num_boxes_boxes.is_static() && scores_ps[0].is_static() && scores_ps[1].is_static() &&
-            has_and_set_equal_bounds(input_value(2))) {
-            const auto num_boxes = num_boxes_boxes.get_length();
-            const auto num_classes = scores_ps[1].get_length();
+        const auto num_boxes = boxes_ps[1].get_max_length();
+        const auto num_classes = scores_ps[1].get_max_length();
+        const auto batch = scores_ps[0].get_max_length();
+        if (num_boxes != -1 && batch != -1 && num_classes != -1 && has_and_set_equal_bounds(input_value(2))) {
             const auto max_output_boxes_per_class = max_boxes_output_from_input();
-
-            out_shape[0] =
-                Dimension(0, std::min(num_boxes, max_output_boxes_per_class) * num_classes * scores_ps[0].get_length());
+            out_shape[0] = Dimension(0, std::min(num_boxes, max_output_boxes_per_class) * num_classes * batch);
```

Comment:
Please add tests for the shape inference if this is a regression

Reply:
it's not